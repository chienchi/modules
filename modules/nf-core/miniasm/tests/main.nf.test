// nf-core modules test miniasm
nextflow_process {

    name "Test Process MINIASM"
    script "../main.nf"
    process "MINIASM"

    tag "modules"
    tag "modules_nfcore"
    tag "miniasm"

    test("bacteroides_fragilis - [fastq.gz, paf]- fasta") {

        when {
            process {
                """                
                input[0] = [ 
                    [ id:'test', single_end:true ], // meta map
                    file(params.test_data['bacteroides_fragilis']['nanopore']['test_fastq_gz'], checkIfExists: true),
                    file(params.test_data['bacteroides_fragilis']['nanopore']['overlap_paf'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.gfa.get(0).get(1)).name).match("output/miniasm/test.gfa.gz") },
                { assert snapshot(file(process.out.assembly.get(0).get(1)).name).match("output/miniasm/test.fasta.gz") },
                { assert snapshot(process.out.versions).match("versions") }
                // Unstable hashes { assert snapshot(process.out).match() }
            )
        }

    }

}
