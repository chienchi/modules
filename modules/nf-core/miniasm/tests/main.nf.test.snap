{
    "versions": {
        "content": [
            [
                "versions.yml:md5,0572fb9059c7518d698bc5d609d63935"
            ]
        ],
        "timestamp": "2023-11-28T14:12:02.729942"
    },
    "output/miniasm/test.gfa.gz": {
        "content": [
            "test.gfa.gz"
        ],
        "timestamp": "2023-11-28T14:12:02.708123"
    },
    "output/miniasm/test.fasta.gz": {
        "content": [
            "test.fasta.gz"
        ],
        "timestamp": "2023-11-28T14:12:02.723547"
    }
}